### ISSUE TITLE
一直看不到春吃？

### ISSUE THREAD

**Issue by kotoricon** [MalitsPlus/ShizuruNotes#37]
无论怎么更新刷新都看不到春吃这个人物，是bug么？

**Comment by MalitsPlus** [comment]
请检查app和数据库是否是最新版本

**Comment by kotoricon** [comment]
> 请检查app和数据库是否是最新版本

刚好那天发完这个不到一两小时再更新数据就有了，之前是一直看不到
卡池刚更新那天确实有数据更新，但是无论怎么搜索就是看不到

**Comment by MalitsPlus** [comment]
你这一说我想起来了，目前笔记的本地数据库版本不是根据数据库文件，而是根据api来判断的。
如果服务端json和数据库文件缓存更新有延时，然后又正好在这个短暂的延时期间内更新了本地数据库的话确实有可能会造成这种现象。
有空我会修改的。

**Comment by kotoricon** [comment]
> 你这一说我想起来了，目前笔记的本地数据库版本不是根据数据库文件，而是根据api来判断的。
> 如果服务端json和数据库文件缓存更新有延时，然后又正好在这个短暂的延时期间内更新了本地数据库的话确实有可能会造成这种现象。
> 有空我会修改的。

试了下今天的新nnk，也是数据有更新然后找不到。。😂

**Comment by MalitsPlus** [comment]
草，这玩意没法改。
我本来以为数据库文件里面会有版本标识能用来判断，结果里面啥也没有（

现状只有不要在收到数据更新推送后立即更新，尽量等一段时间

**Comment by kotoricon** [comment]
> 草，这玩意没法改。
> 我本来以为数据库文件里面会有版本标识能用来判断，结果里面啥也没有（
> 
> 现状只有不要在收到数据更新推送后立即更新，尽量等一段时间

要不你试试问问隔壁初音笔记是怎么解决的？隔壁好像就能瞬间更新

**Comment by MalitsPlus** [comment]
一样的，都是依靠服务端的json来判断。

这个主要是服务端接口和文件同步有延迟造成的问题（或者是cf缓存未及时清空？），所以理论上来说都可能会出现这个bug，只是复现这个bug条件比较苛刻。

我再想下能不能通过数据库文件的hash值来判断。

**Comment by HerDataSam** [comment]
I think the easiest way is to create a 'Force re-download' button.
Or, you can abort the update when the MD5 hash value of the downloaded DB file is different from the JSON hash data. In this case, you should inform it not to confuse the users.

**Comment by MalitsPlus** [comment]
Exactly, i'm considering i should check the hash value after db file download completed. 
But since hash value in `last_version_jp.json` indicates the .cdb file's hash rather than .br file nor .db file, the only thing i can do is check the hash values of the old and new db files, if they are the same, aborts, otherwise accepts, i think.
